libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -77,6 +77,11 @@ internal class MapboxTripSession(
             roadObjects = emptyList()
             routeProgress = null
         }
+
+        isOffRoute = false
+        invalidateLatestBannerInstructionEvent()
+        zLevel = null
```

Comment:
`zLevel` should not be reset, this is tied to the actual position of the car, not related to the route it's traveling on.